### ISSUE TITLE
In battle ranged units move 1 tile before hitting in mele

### ISSUE THREAD

**Issue by krs0** [vcmi/vcmi#1691]
**Describe the bug**
In battle ranged units move 1 tile before hitting in mele

**Expected behavior**
Ranged units should stay put before retaliating.

**Actual behavior**
https://user-images.githubusercontent.com/15194321/225360280-95fa5b4c-7d98-4764-be81-7c4bcf3def10.mp4

**Version**
 - OS: Windows
 - Version: develop

**Comment by IvanSavenko** [comment]
That's AI action, right? AI is permitted to move his units...

**Comment by krs0** [comment]
Yes Battle AI action. But very weird and pointless and  above all, deffinetly not OH3 behavior.

**Comment by dydzio0614** [comment]
current battle ai is not trying to mimic H3 AI

**Comment by rilian-la-te** [comment]
@dydzio0614 and I do not think than it should. I think if we can have smarter battle AI, it will be even better.

**Comment by dydzio0614** [comment]
yep, we can have unlimited alternative ais for another purposes

**Comment by krs0** [comment]
One of VCMI battle AI should mimic exactly! H3 battle AI!

It is the core of H3 battles. Maybe vast majority of players will observe only weird things like this useless movement, but more experienced players rely on battle AI doing things in a certain way.

**Comment by rilian-la-te** [comment]
@krs0 only for neutral creatures and optional. AI players' battle AI should not even try to mimic H3 battle AI. You can do such battle AI (mimic H3) as enhancement, if you want)

**Comment by DjWarmonger** [comment]
>  but more experienced players rely on battle AI doing things in a certain way

Well, they'll have to learn how to play again :P

**Comment by dydzio0614** [comment]
the AI which is supposed to mimic H3 is not started by anybody yet

**Comment by kambala-decapitator** [comment]
> more experienced players rely on battle AI doing things in a certain way

which is called "abusing", if we forget a handful of maps where certain AI behaviour is required

**Comment by krs0** [comment]
At least can we agree that this 1 hex moving is stupid?  :)

**Comment by nullkiller** [comment]
> At least can we agree that this 1 hex moving is stupid? :)

Actually not. You need to prove it somehow affects gameplay in a bad way. Otherwise it just looks a bit strange for human but may have no effect on actual battle result. But even if you find any bad effect this will lead to an additional complexity for battle AI because we will have to analyze this effect. For instance we can check what enemy stacks can reach us on each possible attack hex and select one where it will be less enemies. O less free cells around or whatever else.

**Comment by IvanSavenko** [comment]
Closing, not planned. AI is free to move units in any legal way, StupidAI does not recreates H3 AI
